google/cloud/bigquery_storage_v1/writer.py

```diff
@@ -364,6 +364,291 @@ def _on_rpc_done(self, future):
         thread.start()
 
 
+class Connection(object):
+    def __init__(
+        self,
+        client: big_query_write.BigQueryWriteClient,
+        stream: AppendRowsStream,
+        initial_request_template: gapic_types.AppendRowsRequest,
+        metadata: Sequence[Tuple[str, str]] = (),
+    ) -> None:
+        """A connection abstraction that includes a gRPC connection, a consumer,
+        and a queue.
+
+        Args:
+            client:
+                Client responsible for making requests.
+            stream:
+                The stream that created the connection.
+            initial_request_template:
+                Data to include in the first request sent to the stream. This
+                must contain
+                :attr:`google.cloud.bigquery_storage_v1.types.AppendRowsRequest.write_stream`
+                and
+                :attr:`google.cloud.bigquery_storage_v1.types.AppendRowsRequest.ProtoData.writer_schema`.
+            metadata:
+                Extra headers to include when sending the streaming request.
+        """
+        self._client = client
+        self._stream = stream
+        self._initial_request_template = initial_request_template
+        self._metadata = metadata
+        self._thread_lock = threading.RLock()
+
+        self._rpc = None
+        self._consumer = None
+        self._stream_name = None
+        self._queue = queue.Queue()
+
+        # statuses
+        self._closed = False
+
+    @property
+    def is_active(self) -> bool:
+        """bool: True if this connection is actively streaming.
+
+        Note that ``False`` does not indicate this is complete shut down,
+        just that it stopped getting new messages.
+        """
+        return self._consumer is not None and self._consumer.is_active
+
+    def open(
+        self,
+        initial_request: gapic_types.AppendRowsRequest,
+        timeout: float = _DEFAULT_TIMEOUT,
```

Comment:
will this ever be exposed to the user as a controllable timeout?  I'm fine if it's not, but I do want to make sure we're doing it in a way that complements how users specify such things (defaults vs call-level options, etc).